pkg/trace/api/container_linux.go

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package api
```

Comment:
I'd strongly advise not to re-implement this logic. We do provide a library for that ([see how it's called in dogstatsd](https://github.com/DataDog/datadog-agent/blob/022cc40d129070a41f04627f5a6e2671fe1ee0d8/pkg/dogstatsd/listeners/uds_linux.go#L124)) that handles all configurations, corner cases and caching properly.

Reply:
It looks like the dsd solution will not work for this, as it is specific to `SOCK_DGRAM` type sockets, whereas we are using `SOCK_STREAM` here.